[drogon] build failure

**Host Environment**
- OS: ArchLinux
- Compiler: gcc (GCC) 10.1.0

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install drogon`

**Failure logs** 
```➜ ./vcpkg install drogon
Computing installation plan...
The following packages will be built and installed:
    drogon[core]:x64-linux
Detecting compiler hash for triplet x64-linux...
Starting package 1/1: drogon:x64-linux
Building package drogon[core]:x64-linux...
Could not locate cached archive: /home/valery/.cache/vcpkg/archives/81/81616a4c99a9d142fa1a9c55acfe4bfe0e8fc4e1.zip
-- Using cached /home/valery/Cache/vcpkg/downloads/an-tao-drogon-v1.0.0-beta18.tar.gz
-- Cleaning sources at /home/valery/Cache/vcpkg/buildtrees/drogon/src/0.0-beta18-aa4dfb0262.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/valery/Cache/vcpkg/downloads/an-tao-drogon-v1.0.0-beta18.tar.gz
-- Applying patch vcpkg.patch
-- Applying patch drogon_ctl_install.patch
-- Using source at /home/valery/Cache/vcpkg/buildtrees/drogon/src/0.0-beta18-aa4dfb0262.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: /home/valery/Cache/vcpkg/downloads/tools/cmake-3.17.2-linux/cmake-3.17.2-Linux-x86_64/bin/cmake /home/valery/Cache/vcpkg/buildtrees/drogon/src/0.0-beta18-aa4dfb0262.clean -DBUILD_EXAMPLES=OFF -DCMAKE_MAKE_PROGRAM=/home/valery/Cache/vcpkg/downloads/tools/ninja-1.10.0-linux/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/valery/Cache/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/valery/Cache/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/valery/Cache/vcpkg -D_VCPKG_INSTALLED_DIR=/home/valery/Cache/vcpkg/installed -DVCPKG_MANIFEST_INSTALL=OFF -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/valery/Cache/vcpkg/packages/drogon_x64-linux
    Working Directory: /home/valery/Cache/vcpkg/buildtrees/drogon/x64-linux-rel
    Error code: 1
    See logs for more information:
      /home/valery/Cache/vcpkg/buildtrees/drogon/config-x64-linux-rel-out.log
      /home/valery/Cache/vcpkg/buildtrees/drogon/config-x64-linux-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:319 (vcpkg_execute_required_process)
  ports/drogon/portfile.cmake:12 (vcpkg_configure_cmake)
  scripts/ports.cmake:79 (include)


Error: Building package drogon:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: drogon:x64-linux
  Vcpkg version: 2020.06.15-unknownhash
```
In the `config-x64-linux-rel-err.log`
```
CMake Error at cmake_modules/Findpg.cmake:16 (set_target_properties):
  Property INTERFACE_LINK_LIBRARIES may not contain link-type keyword
  "optimized".  The INTERFACE_LINK_LIBRARIES property may contain
  configuration-sensitive generator-expressions which may be used to specify
  per-configuration rules.
Call Stack (most recent call first):
  /home/valery/Cache/vcpkg/scripts/buildsystems/vcpkg.cmake:455 (_find_package)
  CMakeLists.txt:164 (find_package
```

--- a/ports/drogon/CONTROL
+++ b/ports/drogon/CONTROL
@@ -1,5 +1,5 @@
 Source: drogon
-Version: v1.0.0-beta18
+Version: v1.0.0-beta20
 Homepage: https://github.com/an-tao/drogon
 Description:Drogon: A C++14/17 based HTTP web application framework running on Linux/macOS/Unix/Windows
 Build-Depends: trantor, zlib, jsoncpp, libmariadb, libpq, sqlite3, brotli

--- a/ports/drogon/pg.patch
+++ b/ports/drogon/pg.patch
@@ -1,0 +1,27 @@
+diff --git a/cmake_modules/Findpg.cmake b/cmake_modules/Findpg.cmake
+index 53037f2..d13f384 100644
+--- a/cmake_modules/Findpg.cmake
++++ b/cmake_modules/Findpg.cmake
+@@ -4,20 +4,14 @@
+ #
+ # This module defines PG_INCLUDE_DIRS, where to find header, etc. PG_LIBRARIES,
+ # the libraries needed to use PostgreSQL. pg_FOUND, If false, do not try to use
+-# PostgreSQL.
+-# pg_lib - The imported target library.
++# PostgreSQL. pg_lib - The imported target library.
+ 
+ find_package(PostgreSQL)
+ if(PostgreSQL_FOUND)
+   set(PG_LIBRARIES ${PostgreSQL_LIBRARIES})
+   set(PG_INCLUDE_DIRS ${PostgreSQL_INCLUDE_DIRS})
+-  message(STATUS "pg inc: " ${PostgreSQL_INCLUDE_DIRS})
+   add_library(pg_lib INTERFACE IMPORTED)
+-  set_target_properties(pg_lib
+-                        PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
+-                                   "${PostgreSQL_INCLUDE_DIRS}"
+-                                   INTERFACE_LINK_LIBRARIES
+-                                   "${PostgreSQL_LIBRARIES}")
++  target_link_libraries(pg_lib INTERFACE PostgreSQL::PostgreSQL)
+   mark_as_advanced(PG_INCLUDE_DIRS PG_LIBRARIES)
+ endif(PostgreSQL_FOUND)
+ include(FindPackageHandleStandardArgs)

--- a/ports/drogon/portfile.cmake
+++ b/ports/drogon/portfile.cmake
@@ -5,9 +5,9 @@ vcpkg_from_github(
     SHA512 10da2765df24848964358ea3f6a8843664d142bfd1d276649438c5d79ad88f98467ac742d3ff23e9fda93b89ab9568b2ecea8c3ee4d527d0b6b4a5892025e24e
     HEAD_REF master
     PATCHES
-    vcpkg.patch
-    drogon_ctl_install.patch
-    pg.patch
+        vcpkg.patch
+        drogon_ctl_install.patch
+        pg.patch
 )
 
 vcpkg_configure_cmake(
@@ -20,7 +20,7 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
 
 # Fix CMake files
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/drogon)
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
 
 # # Remove includes in debug
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")

--- a/ports/drogon/portfile.cmake
+++ b/ports/drogon/portfile.cmake
@@ -1,12 +1,13 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO an-tao/drogon
-    REF v1.0.0-beta18
-    SHA512 38d57e6b5cc43bd1ac07c980453dda1f4c23a8c78eca942dd531d20661d7f088fc0a3d0f4e7cedddf98bac5a32be0330911cbf839d4956e05ee0265a68c7faa6
+    REF v1.0.0-beta20
+    SHA512 10da2765df24848964358ea3f6a8843664d142bfd1d276649438c5d79ad88f98467ac742d3ff23e9fda93b89ab9568b2ecea8c3ee4d527d0b6b4a5892025e24e
     HEAD_REF master
     PATCHES
     vcpkg.patch
     drogon_ctl_install.patch
+    pg.patch
 )
 
 vcpkg_configure_cmake(
@@ -24,6 +25,9 @@ vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/drogon)
 # # Remove includes in debug
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+endif()
 # Handle copyright
 file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
 

